### ISSUE TITLE
Adaption to new imagemagick version

### ISSUE THREAD

**Issue by snail123815** [giacmir/Inkscape-JPEG-export-extension#14]
"convert" command is no longer working if one installed the newest version of imagemagick.
Instead, "magick" should be used instead.

The only thing needed to do after moved the two files is to change the *.py file, change the command.
Works in windows 10 64bit, ImageMagick-7.0.3-Q16 installed with legacy not included.

line 159:
`        command = "magick -quality %s -density %s \"%sjpinkexp.png\" \"%s\" " % (self.options.quality, self.options.density, tmp, outfile)
`

**Comment by Moini** [comment]
For an explanation, see also: 
http://stackoverflow.com/questions/4481573/how-to-override-windows-convert-command-by-imagemagicks-one
and 
https://www.imagemagick.org/script/porting.php, section 'command changes'

Both commands should work on Windows, though, you'd only need to change your path variables.

But to have it as comfortable as possible, I'd suggest to add this: 

if imagemagick version >= 7.0, then use 'magick' instead of 'convert' (convert will still work, though).

**Comment by giacmir** [comment]
The best way is to check against the imagemagick version and use the appropriate command. Changing the command directly will make the extension unusable to people having older versions (like me eh eh).

I think that this can be done by invoking the magick command with something like `magick -v` and see if the system responds with an error. If it does then use `convert`, else `magick`.

**Comment by Moini** [comment]
What kind of troubles does it cause to enter an invalid command into a Windows 'shell' from an Inkscape extension? (sorry, I really don't know, but it would be good if we were able to prevent that...)

**Comment by leycec** [comment]
> I think that this can be done by invoking the magick command with something like magick -v and see if the system responds with an error.

**Don't do this.** Just test for the existence of executable files with basenames `convert` and `magick` (*in that order*) in the current `${PATH}`. This is trivially implemented in cross-platform pure Python as follows:

```
import shutil

# Absolute path of the ImageMagick conversion command in the current ${PATH}.
magick_filename = shutil.which('convert') or shutil.which('magick')

# If no such command exists, raise an exception.
if magick_filename is None:
    raise ValueError(
        'No ImageMagick conversion command found '
        '(i.e., neither "convert" nor "magick" in ${PATH}).')
```

**Comment by giacmir** [comment]
Thank you @leycec for your suggestion.

**Comment by jimgregory** [comment]
I'm currently using the latest version of Debian (Stretch), and it does not support the 'magick' command in newer versions of Imagemagick, only the older 'convert' and 'mogrify'.  So, I've added a test to see which version of Imagemagick is installed and runs the appropriate command, and created a pull request to incorporate it.

It should be cross-platform based on the Python library I used, but I don't have a Windows or MacOS box to confirm this.
